quattor/blockdevices.pan

```diff
@@ -53,6 +53,7 @@ type blockdevices_md_type = {
     "raid_level" : string with match (SELF,'^RAID[0156]$')
     "stripe_size" : long = 64 # "Stripe size in KB"
     "num_spares" ? long # "Number of spare devices"
+    "metadata" : string = '0.90' with match(SELF, '^(0(.90)?|1(.[012])?|ddf|imsm)$')
```

Comment:
The dots needs to be escaped. This appears to be matching the spec from http://linux.die.net/man/8/mdadm which says that the default is 1.2 _not_ 0.90. Perhaps this should be made optional instead of defaulting to an older version?

Using a regexp to fake an ENUM type is pretty ugly. How about using a list instead?
```with index(SELF, list('0', '0.90', '1.0', '1.1', '1.2', 'ddf', 'imsm')) > 0;``` That probably results in a user-unfriendly error message so still not idea.